person.rb

```diff
@@ -1,22 +1,34 @@
 # Person class
-class Person
-  attr_reader :id
+require_relative 'nameable'
+
+class Person < Nameable
+  attr_reader :id, :rental
   attr_accessor :name, :age
 
-  def initialize(age, parent_permission: true, name: 'Unknown')
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..1000)
-    @name = name
     @age = age
+    @name = name
     @parent_permission = parent_permission
   end
 
   def can_use_services?
     is_of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     @age >= 18
   end
+
+  def add_rental(rental)
+    @rentals << rental
+    self
+  end
```

Comment:
- [ ] This method suppose to add a new rental and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association, Please fix that.
[hint] It will be something like that:
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
 end
```